### DISCUSSION TITLE
Unable to create tunnel with when application upgraded strongswan from 5.9.6 to 5.9.10

### DISCUSSION THREAD

**Discussion by hbhokare** [strongswan/strongswan#2078] [Q&A]
We have developed the IPSec-based application and used different strongswan/davici APIs from our application. Earlier our application was using the strongswan 5.9.6 on Linux kernel 4.18 now we have upgraded the kernel to 5.14 and strongswan to 5.9.10. with this change, the application is unable to create a secured tunnel.  We also tried to create the tunnel without our application code and using Linux host with the below-provided swanctl.conf file on newly updated strongswan and kernel still failed to create a tunnel same config file successfully creates the secured tunnel. 

Site A:
====

connections {
         version = 2
         mobike = no
         proposals = default
           host-host {
               remote_addrs = A.B.111.100
               local_addrs = A.B.111.102
               local {
                   auth=psk
                   id=A.B.111.102
               }
               remote {
                   auth =psk
                   id=A.B.111.100
               }
               children {
                   net-net {
                       start_action = trap
                       esp_proposals = null
                       mode = tunnel
                   }
               }
           }
}
secrets {
    ike-host-host {
        secret = abcdxxxx
        id-1 = A.B.111.102
        id-2 = A.B.111.100
    }
}

Site B:
====

connections {
         version = 2
         mobike = no
         proposals = default
           host-host {
               remote_addrs = A.B.111.102
               local_addrs = A.B.111.100
               local {
                   auth=psk
                   id=A.B.111.100
               }
               remote {
                   auth =psk
                   id=A.B.111.102
               }
               children {
                   net-net {
                       start_action = trap
                       esp_proposals = null
                       mode = tunnel
                   }
               }
           }
}
secrets {
    ike-host-host {
        secret = abcdxxxx
        id-1 = A.B.111.100
        id-2 = A.B.111.102
    }
}

Here is IKE initiate failure we are getting when we initiate from one side

[00:43:27] fab3node2:/dumps # swanctl -i -i host-host --debug 4
plugin 'sqlite': failed to load - sqlite_plugin_create not found and no plugin file available
created thread 01 [2131037]
started worker thread 01
no events, waiting
created thread 02 [2131036]
started worker thread 02
  watching 7 for reading
watcher going to poll() 2 fds
watched FD 7 ready to read
watcher going to poll() 1 fds
created thread 03 [2131035]
started worker thread 03
watcher got notification, rebuilding
  watching 7 for reading
watcher going to poll() 2 fds
watcher got notification, rebuilding
  watching 7 for reading
watcher going to poll() 2 fds
created thread 04 [2131034]
started worker thread 04
watched FD 7 ready to read
watcher going to poll() 1 fds
[IKE] retransmit 1 of request with message ID 0
watcher got notification, rebuilding
  watching 7 for reading
watcher going to poll() 2 fds
watched FD 7 ready to read
watcher going to poll() 1 fds
[NET] sending packet: from 192.168.111.102 to 192.168.111.100[500] (1048 bytes)
watcher got notification, rebuilding
  watching 7 for reading
watcher going to poll() 2 fds
watched FD 7 ready to read
watcher going to poll() 1 fds
watcher got notification, rebuilding
initiate failed: establishing IKE_SA 'host-host' failed

**Comment by hbhokare** [comment]
Site A charon log 
=========
Wed Jan 24 00:43:25 2024 16[CFG] vici initiate IKE_SA 'host-host'
Wed Jan 24 00:43:25 2024 10[IKE] <host-host|7> initiating IKE_SA host-host[7] to 192.168.111.100
Wed Jan 24 00:43:39 2024 15[CFG] vici initiate IKE_SA 'host-host'
Wed Jan 24 00:43:40 2024 06[IKE] <host-host|7> retransmit 1 of request with message ID 0
Wed Jan 24 00:43:55 2024 10[IKE] <host-host|7> retransmit 2 of request with message ID 0
Wed Jan 24 00:44:10 2024 04[IKE] <host-host|7> giving up after 2 retransmits
Wed Jan 24 00:44:10 2024 04[IKE] <host-host|7> establishing IKE_SA failed, peer not responding
Wed Jan 24 00:44:10 2024 16[CFG] vici connection 17 unknown
Wed Jan 24 00:52:45 2024 15[CFG] loaded IKE shared key with id 'ike-host-host' for: '192.168.111.102', '192.168.111.100'
Wed Jan 24 00:52:45 2024 06[CFG] updated vici connection: host-host
Wed Jan 24 00:52:45 2024 06[CFG] uninstalling 'net-net'
Wed Jan 24 00:52:45 2024 06[CFG] installing 'net-net'
Wed Jan 24 00:53:14 2024 16[CFG] vici initiate IKE_SA 'host-host'
Wed Jan 24 00:53:14 2024 15[IKE] <host-host|8> initiating IKE_SA host-host[8] to 192.168.111.100
Wed Jan 24 00:53:29 2024 09[IKE] <host-host|8> retransmit 1 of request with message ID 0
Wed Jan 24 00:53:44 2024 15[IKE] <host-host|8> retransmit 2 of request with message ID 0
Wed Jan 24 00:53:59 2024 12[IKE] <host-host|8> giving up after 2 retransmits
Wed Jan 24 00:53:59 2024 12[IKE] <host-host|8> establishing IKE_SA failed, peer not responding



Site B Charon log 
===========
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6> initiating IKE_SA host-host[6] to 192.168.111.102
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6> IKE_SA host-host[6] state change: CREATED => CONNECTING
Wed Jan 24 00:53:23 2024 13[CFG] <host-host|6> configured proposals: IKE:AES_CBC_128/AES_CBC_192/AES_CBC_256/AES_CTR_128/AES_CTR_192/AES_CTR_256/CAMELLIA_CBC_128/CAMELLIA_CBC_192/CAMELLIA_CBC_256/CAMELLIA_CTR_128/CAMELLIA_CTR_192/CAMELLIA_CTR_256/3DES_CBC/HMAC_SHA2_256_128/HMAC_SHA2_384_192/HMAC_SHA2_512_256/AES_XCBC_96/AES_CMAC_96/HMAC_SHA1_96/PRF_HMAC_SHA2_256/PRF_HMAC_SHA2_384/PRF_HMAC_SHA2_512/PRF_AES128_XCBC/PRF_AES128_CMAC/PRF_HMAC_SHA1/ECP_256/ECP_384/ECP_521/CURVE_25519/CURVE_448/NEWHOPE_128/MODP_3072/MODP_4096/MODP_6144/MODP_8192/MODP_2048, IKE:AES_CCM_16_128/AES_CCM_16_192/AES_CCM_16_256/AES_GCM_16_128/AES_GCM_16_192/AES_GCM_16_256/CHACHA20_POLY1305/CAMELLIA_CCM_16_128/CAMELLIA_CCM_16_192/CAMELLIA_CCM_16_256/AES_CCM_8_128/AES_CCM_8_192/AES_CCM_8_256/AES_CCM_12_128/AES_CCM_12_192/AES_CCM_12_256/AES_GCM_8_128/AES_GCM_8_192/AES_GCM_8_256/AES_GCM_12_128/AES_GCM_12_192/AES_GCM_12_256/CAMELLIA_CCM_8_128/CAMELLIA_CCM_8_192/CAMELLIA_CCM_8_256/CAMELLIA_CCM_12_128/CAMELLIA_CCM_12_192/CAMELLIA_CCM_12_256/PRF_HMAC_SHA2_256/PRF_HMAC_SHA2_384/PRF_HMAC_SHA2_512/PRF_AES128_XCBC/PRF_AES128_CMAC/PRF_HMAC_SHA1/ECP_256/ECP_384/ECP_521/CURVE_25519/CURVE_448/NEWHOPE_128/MODP_3072/MODP_4096/MODP_6144/MODP_8192/MODP_2048
Wed Jan 24 00:53:23 2024 13[CFG] <host-host|6> sending supported signature hash algorithms: sha256 sha384 sha512 identity
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6> natd_chunk => 22 bytes @ 0x7fc508079ee0
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6>    0: 7C D9 74 5E F3 45 90 9B 00 00 00 00 00 00 00 00  |.t^.E..........
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6>   16: C0 A8 6F 64 00 00                                ..od..
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6> natd_hash => 20 bytes @ 0x7fc508079f20
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6>    0: 12 FE EF CE B1 BF 34 2F 72 61 56 08 1A CD 82 AB  ......4/raV.....
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6>   16: 3B E3 C2 72                                      ;..r
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6> natd_chunk => 22 bytes @ 0x7fc508079f00
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6>    0: 7C D9 74 5E F3 45 90 9B 00 00 00 00 00 00 00 00  |.t^.E..........
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6>   16: C0 A8 6F 66 01 F4                                ..of..
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6> natd_hash => 20 bytes @ 0x7fc508069550
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6>    0: 6C 25 A8 6B B1 0D CF C3 56 62 86 55 37 23 02 E1  l%.k....Vb.U7#..
Wed Jan 24 00:53:23 2024 13[IKE] <host-host|6>   16: BC FE 45 92                                      ..E.
Wed Jan 24 00:53:38 2024 11[IKE] <host-host|6> retransmit 1 of request with message ID 0
Wed Jan 24 00:53:53 2024 05[IKE] <host-host|6> retransmit 2 of request with message ID 0
Wed Jan 24 00:54:08 2024 05[IKE] <host-host|6> giving up after 2 retransmits
Wed Jan 24 00:54:08 2024 05[IKE] <host-host|6> establishing IKE_SA failed, peer not responding
Wed Jan 24 00:54:08 2024 05[IKE] <host-host|6> IKE_SA host-host[6] state change: CONNECTING => DESTROYING
Wed Jan 24 00:54:08 2024 06[CFG] vici client 22 disconnected


We have also verified the firewall rules on the system they are set up as follows.

    iptables -A INPUT -m state --state NEW -m udp -p udp --dport 500 -j ACCEPT
    iptables -A INPUT -m state --state NEW -m tcp -p tcp --dport 4500 -j ACCEPT
    iptables -A INPUT -m state --state NEW -m udp -p udp --dport 4500 -j ACCEPT
    iptables -A INPUT -p 50 -j ACCEPT

**Reply by tobiasbrunner** [reply]
Those rules are definitely incomplete (but whether that's a problem depends on the default firewall policy for INPUT) and strongSwan currently only uses UDP ports 500 and 4500, not TCP.

Anyway, there is apparently some connectivity issue you have to figure out (could be a config problem, a firewall problem, a general networking issue, etc.).

**Comment by hbhokare** [comment]
In this case, the network is reachable when strongswan conf is not installed as soon as we install the config using the following config files it stops pinging the peer and we see peer not responding error on a node that is initiating the tunnel. does config install alter the firewall rules or anything from networking side.

swanctl.conf
========

connections {
         version = 2
         mobike = no
         proposals = aes256-sha384-ecp384
           host-host {
               remote_addrs = 192.168.111.100
               local_addrs = 192.168.111.102
               local {
                   auth=psk
                   id=192.168.111.102
               }
               remote {
                   auth=psk
                   id=192.168.111.100
               }
               children {
                   net-net {
                       start_action = trap
                       esp_proposals = aes256-sha256-ecp256
                       mode = tunnel
                   }
               }
           }
}
secrets {
    ike-host-host {
        secret = abcd1234
        id-1 = 192.168.111.102
        id-2 = 192.168.111.100
    }
}
